packages/core/src/readers/SimpleDirectoryReader.edge.ts

```diff
@@ -48,67 +56,99 @@ export class SimpleDirectoryReader implements BaseReader {
       fs = defaultFS,
       defaultReader = new TextFileReader(),
       fileExtToReader,
+      numWorkers = 1,
+      overrideReader,
     } = params;
 
+    if (numWorkers < 1 || numWorkers > 9) {
+      throw new Error("The number of workers must be between 1 - 9.");
+    }
+
     // Observer can decide to skip the directory
     if (
       !this.doObserverCheck("directory", directoryPath, ReaderStatus.STARTED)
     ) {
       return [];
     }
 
-    const docs: Document[] = [];
+    const filePathQueue: string[] = [];
+
     for await (const filePath of walk(fs, directoryPath)) {
-      try {
-        const fileExt = path.extname(filePath).slice(1).toLowerCase();
+      filePathQueue.push(filePath);
+    }
 
-        // Observer can decide to skip each file
-        if (!this.doObserverCheck("file", filePath, ReaderStatus.STARTED)) {
-          // Skip this file
-          continue;
-        }
+    const processFilesParams: ProcessFilesParams = {
+      fs,
+      defaultReader,
+      fileExtToReader,
+      overrideReader,
+    };
 
-        let reader: BaseReader;
+    const limit = pLimit(numWorkers);
+    const workerPromises = filePathQueue.map((filePath) =>
+      limit(() => this.processFiles(filePath, processFilesParams)),
```

Comment:
isn't it processing just one file?
```suggestion
      limit(() => this.processFile(filePath, processFilesParams)),
```